### PR TITLE
[feat]: [CDS-40800]: Added skelton files

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Describe your changes
Added basic code to add yaml save code google artifact

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35669)

### CODE REVIEW

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by vivekkarnatiharness** [comment]
trigger ti1

**Comment by vivekkarnatiharness** [comment]
trigger ti0

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by vivekkarnatiharness** [comment]
trigger ti0

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 20 Issues 
* Code issue 19 Code issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35669)

**Review by shivamnegi94 — CHANGES_REQUESTED** [review]
> Reviewed 14 of 32 files at r1, 34 of 34 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35669)* status: 1 change requests. 0 of 1 approvals obtained. Waiting on team(s): sto. 

___
*[`400-rest/src/main/java/software/wings/app/WingsModule.java` line 1079 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9e_1umBXroVDRmgl1T:-N9e_1un41HdzbDJcifL:b3wox2t) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/400-rest/src/main/java/software/wings/app/WingsModule.java#L1079)):*
> ```Java
>     bind(EcrClassicService.class).to(EcrClassicServiceImpl.class);
>     bind(GcrApiService.class).to(GcrApiServiceImpl.class);
>     bind(GarApiService.class).to(GARApiServiceImpl.class);
> ```

This class is even  in NGmodule also, is it needed here ?
___
*[`400-rest/src/test/java/software/wings/helpers/ext/gar/GarApiServiceTest.java` line 47 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9e_Zdw8Y7F2q-9o_YQ:-N9e_Zdw8Y7F2q-9o_YR:b-oqt8ne) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/400-rest/src/test/java/software/wings/helpers/ext/gar/GarApiServiceTest.java#L47)):*
> ```Java
>   //
>   //         wireMockRule.stubFor(WireMock.get(WireMock.urlEqualTo("/v2/teapot/tags/list"))
>   //                 .withHeader("Authorization", equalTo("auth"))
> ```

Why these lines are commented ?
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java` line 62 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9ea4Ss3rp5Vqpk_CjQ:-N9ea4Ss3rp5Vqpk_CjR:b5nv62r) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java#L62)):*
> ```Java
>         default:
>           saveLogs(executionLogCallback,
>               "No corresponding Docker artifact task type [{}]: " + artifactTaskParameters.toString());
> ```

please update the logs line docker to GAR
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java` line 66 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9ea9T575nvxFVcofM5:-N9ea9T575nvxFVcofM6:bs6qvsh) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java#L66)):*
> ```Java
>           return ArtifactTaskResponse.builder()
>               .commandExecutionStatus(CommandExecutionStatus.FAILURE)
>               .errorMessage("There is no Docker artifact task type impl defined for - "
> ```

Here as well

**Comment by vivekkarnatiharness** [comment]
*[`400-rest/src/main/java/software/wings/app/WingsModule.java` line 1079 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9e_1umBXroVDRmgl1T:-N9frkgR5nWt5Qxd4vGK:b-dm8j3m) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/400-rest/src/main/java/software/wings/app/WingsModule.java#L1079)):*
<details><summary><i>Previously, shivamnegi94 (Shivam Negi) wrote…</i></summary><blockquote>

This class is even  in NGmodule also, is it needed here ?
</blockquote></details>

Yes,It should be in Wingmodule as well!

**Comment by vivekkarnatiharness** [comment]
*[`400-rest/src/test/java/software/wings/helpers/ext/gar/GarApiServiceTest.java` line 47 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9e_Zdw8Y7F2q-9o_YQ:-N9frtSY0MRcPRGJe7TO:ba3dfob) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/400-rest/src/test/java/software/wings/helpers/ext/gar/GarApiServiceTest.java#L47)):*
<details><summary><i>Previously, shivamnegi94 (Shivam Negi) wrote…</i></summary><blockquote>

Why these lines are commented ?
</blockquote></details>

Removed

**Comment by vivekkarnatiharness** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java` line 62 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9ea4Ss3rp5Vqpk_CjQ:-N9frzoa1sRF6kJyLsBP:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java#L62)):*
<details><summary><i>Previously, shivamnegi94 (Shivam Negi) wrote…</i></summary><blockquote>

please update the logs line docker to GAR
</blockquote></details>

Done

**Comment by vivekkarnatiharness** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java` line 66 at r2](https://reviewable.io/reviews/harness/harness-core/35669#-N9ea9T575nvxFVcofM5:-N9fs0aOE30aVMC-cUPq:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/99ef77f5a3599767e2b904b0f1af4abbee035346/930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHelper.java#L66)):*
<details><summary><i>Previously, shivamnegi94 (Shivam Negi) wrote…</i></summary><blockquote>

Here as well
</blockquote></details>

Done.

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35669)* status: 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. Waiting on team(s): sto.

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35669)* status: 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. Waiting on team(s): sto.

**Review by shivamnegi94 — APPROVED** [review]

File: 125-cd-nextgen/src/test/java/io/harness/cdng/artifact/resources/gar/service/GarResourceServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import org.mockito.Spy;
+
+@OwnedBy(HarnessTeam.CDC)
+public class GarResourceServiceImplTest extends CategoryTest {
+  private static final String ACCOUNT_ID = "accountId";
+  private static final String REGION = "imagePath";
+  private static final String repositoryName = "repositoryName";
+  private static final String project = "project";
+  private static final String version = "version";
+  private static final String pkg = "pjg";
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/test/java/io/harness/cdng/artifact/resources/gar/service/GarResourceServiceImplTest.java:54]: Fix Typo

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/GarArtifactSummary.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import lombok.Builder;
+import lombok.Data;
+
+@OwnedBy(CDC)
+@Data
+@Builder
+@JsonTypeName(ArtifactSourceConstants.GOOGLE_ARTIFACT_REGISTRY_NAME)
+public class GarArtifactSummary implements ArtifactSummary {
+  String pkg;
```
Comment (puthrayaharness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/GarArtifactSummary.java:17]: @JsonProperty("package")

Comment (puthrayaharness) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/GarArtifactSummary.java:17]: Summary should also include project, repo & region

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/artifacts/ArtifactDelegateRequestUtils.java [hunk 3]
```diff
@@ -72,6 +74,22 @@ public EcrArtifactDelegateRequest getEcrDelegateRequest(String imagePath, String
         .sourceType(sourceType)
         .build();
   }
+  public GarDelegateRequest getGoogleArtifactDelegateRequest(String region, String repositoryName, String project,
+      String pkg, String version, String versionRegex, GcpConnectorDTO gcpConnectorDTO,
+      List<EncryptedDataDetail> encryptedDataDetails, ArtifactSourceType sourceType, int MAXBUILDS) {
```
Comment (puthrayaharness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/artifacts/ArtifactDelegateRequestUtils.java:79]: `MAXBUILDS` -> `maxBuilds`

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/outcome/GarArtifactOutcome.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  String versionRegex;
+  String type;
+  String identifier;
+  String image;
+  String registryHostname;
+  String imagePullSecret;
+  boolean primaryArtifact;
+  @Override
+  public ArtifactSummary getArtifactSummary() {
+    return null;
```
Comment (puthrayaharness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/outcome/GarArtifactOutcome.java:39]: Why is this `null`?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHandler.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+@OwnedBy(HarnessTeam.CDC)
+@Singleton
+@AllArgsConstructor(access = AccessLevel.PACKAGE, onConstructor = @__({ @Inject }))
+@Slf4j
+public class GARArtifactTaskHandler extends DelegateArtifactTaskHandler<GarDelegateRequest> {
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/googleartifactregistry/GARArtifactTaskHandler.java:41]: Please add test cases to this file. We should cover cases where we are throwing exceptions.

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    OkHttpClient okHttpClient = Http.getOkHttpClient(url, garinternalConfig.isCertValidationRequired());
+    Retrofit retrofit = new Retrofit.Builder()
+                            .client(okHttpClient)
+                            .baseUrl(url)
+                            .addConverterFactory(JacksonConverterFactory.create())
+                            .build();
+    return retrofit.create(GarRestClient.class);
+  }
+  public String getUrl() {
+    return "https://"
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:57]: Make this oneline.

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GarApiService.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+import io.harness.artifacts.beans.BuildDetailsInternal;
+import io.harness.artifacts.gar.beans.GarInternalConfig;
+
+import java.util.List;
+
+@OwnedBy(HarnessTeam.CDC)
+public interface GarApiService {
+  int MAX_NO_OF_TAGS_PER_IMAGE = 10000;
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GarApiService.java:12]: Remove this.

File: 980-commons/src/main/java/io/harness/artifact/ArtifactMetadataKeys.java [hunk 8]
```diff
@@ -18,4 +18,5 @@ public interface ArtifactMetadataKeys {
   String artifactPath = "artifactPath";
   String artifactName = "artifactName";
   String artifactFileSize = "artifactFileSize";
+  String PACKAGE = "package";
```
Comment (puthrayaharness) [inline 980-commons/src/main/java/io/harness/artifact/ArtifactMetadataKeys.java:21]: `PACKAGE` -> `artifactPackage`. 
Where is this used? We can change `package` to `artifactPackage`

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+  private List<BuildDetailsInternal> processPage(GarPackageVersionResponse tagsPage, String versionRegex) {
+    if (tagsPage != null && EmptyPredicate.isNotEmpty(tagsPage.getTags())) {
+      int index = tagsPage.getTags().get(0).getName().lastIndexOf("/");
+      List<BuildDetailsInternal> buildDetails =
+          tagsPage.getTags()
+              .stream()
+              .map(tag -> {
+                String tagFinal = tag.getName().substring(index + 1);
+                Map<String, String> metadata = new HashMap();
+                metadata.put(ArtifactMetadataKeys.PACKAGE, tagFinal);
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:161]: Also add repository name, project name etc to the this?

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 960-api-services/src/test/java/io/harness/gar/GARApiServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+  @Test
+  @Owner(developers = vivekveman)
+  @Category(UnitTests.class)
+  public void verifyBuildNumberTest() {
+    BuildDetailsInternal actual = garApiServiceImpl.verifyBuildNumber(gcpInternalConfig, "latest10");
+    assertThat(actual.getNumber()).isEqualTo("latest10");
+  }
+  //  @Test
+  //  @Owner(developers = vivekveman)
```
Comment (puthrayaharness) [inline 960-api-services/src/test/java/io/harness/gar/GARApiServiceTest.java:126]: If these changes are not requires Please remove it.

File: 120-ng-manager/config.yml [hunk 11]
```diff
@@ -265,7 +265,7 @@ pmsGrpcClientConfig:
   target: localhost:12011
   authority: localhost
 
-shouldConfigureWithPMS: ${SHOULD_CONFIGURE_WITH_PMS:-false}
```
Comment (puthrayaharness) [inline 120-ng-manager/config.yml:268]: Please revert

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by vivekkarnatiharness** [comment]
trigger ti0

**Comment by vivekkarnatiharness** [comment]
trigger ti0

**Comment by vivekkarnatiharness** [comment]
trigger ti0

**Comment by vivekkarnatiharness** [comment]
trigger ti1
trigger ti2

**Comment by vivekkarnatiharness** [comment]
trigger ti1

**Comment by vivekkarnatiharness** [comment]
trigger ti4

**Comment by vivekkarnatiharness** [comment]
trigger ti3

**Comment by vivekkarnatiharness** [comment]
trigger ti4

**Comment by vivekkarnatiharness** [comment]
trigger ti4

**Comment by vivekkarnatiharness** [comment]
trigger ti0

**Comment by vivekkarnatiharness** [comment]
trigger ti4

**Comment by vivekkarnatiharness** [comment]
trigger ft1

**Comment by vivekkarnatiharness** [comment]
trigger ft2

**Comment by vivekkarnatiharness** [comment]
trigger feature-build

**Comment by vivekkarnatiharness** [comment]
trigger feature-build

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/mappers/GarRequestResponseMapper.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+import io.harness.artifacts.gar.beans.GarInternalConfig;
+import io.harness.delegate.task.artifacts.ArtifactSourceType;
+import io.harness.delegate.task.artifacts.gar.GarDelegateRequest;
+import io.harness.delegate.task.artifacts.gar.GarDelegateResponse;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class GarRequestResponseMapper {
+  public GarInternalConfig toGarInternalConfig(GarDelegateRequest request, String bearerToken) {
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/mappers/GarRequestResponseMapper.java:13]: These can be static methods

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/mappers/GarRequestResponseMapper.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    return GarInternalConfig.builder()
+        .bearerToken(bearerToken)
+        .project(request.getProject())
+        .repositoryName(request.getRepositoryName())
+        .region(request.getRegion())
+        .pkg(request.getPkg())
+        .maxBuilds(request.getMaxBuilds())
+        .build();
+  }
+  public GarDelegateResponse toGarResponse(BuildDetailsInternal buildDetailsInternal, GarDelegateRequest request) {
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/mappers/GarRequestResponseMapper.java:23]: can be static method

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+      case 404:
+        throw NestedExceptionUtils.hintWithExplanationException(
+            "Please Check Project,RepositoryName,Package,Region fields",
+            "Google Artifact Registry: Check Values of Project,RepositoryName,Package,Region Field",
+            new InvalidArtifactServerException(response.errorBody().toString(), USER));
+      case 400:
+        return false;
+      case 401:
+        throw NestedExceptionUtils.hintWithExplanationException(
+            "Google Artifact Registry: Connector provided Is not Having Artifact Registry Reader permission",
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:174]: we are capitalizing message unnecessarily- `Is` -> `is`

Comment (puthrayaharness) [reply 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:174]: -> `The connector provided does not have sufficient privileges to access Google artifact registry`

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+            "Check connector's permission and credentials",
+            new InvalidArtifactServerException(response.errorBody().toString(), USER));
+      default:
+        throw NestedExceptionUtils.hintWithExplanationException("Google Artifact Registry", "Google Artifact Registry",
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:178]: Please fix the hint & explanation.

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+    if (response.isSuccessful()) {
+      return true;
+    }
+
+    log.error("Request not successful. Reason: {}", response);
+    int code = response.code();
+    switch (code) {
+      case 404:
+        throw NestedExceptionUtils.hintWithExplanationException(
+            "Please Check Project,RepositoryName,Package,Region fields",
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:167]: You should have space after commas. Unnecessary capitalization

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+            "Google Artifact Registry: Check Values of Project,RepositoryName,Package,Region Field",
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:168]: Both Hint & explanation are the same. Can you please ensure they are different?

File: 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java [hunk 18]
```diff
@@ -0,0 +1,223 @@
...
+            new InvalidArtifactServerException(response.errorBody().toString(), USER));
+      case 400:
+        return false;
+      case 401:
+        throw NestedExceptionUtils.hintWithExplanationException(
+            "Google Artifact Registry: Connector provided Is not Having Artifact Registry Reader permission",
+            "Check connector's permission and credentials",
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java:175]: Check -> Please check

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by puthrayaharness** [comment]
*[`960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java` line 178 at r9](https://reviewable.io/reviews/harness/harness-core/35669#-N8OxTjoQbm4TUzwkRku-r9-178:-NAhyqIqDUiZ59jH1KHN:b-r7jn8o) ([raw file](https://github.com/harness/harness-core/blob/53d2ccf66e99de40e05299a60357ea3895451c8f/960-api-services/src/main/java/io/harness/artifacts/gar/service/GARApiServiceImpl.java#L178)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Please fix the hint & explanation. 
</blockquote></details>

Could you please check this?

**Comment by vivekkarnatiharness** [comment]
trigger codebasehash

**Comment by vivekkarnatiharness** [comment]
trigger codebasehash

**Comment by vivekkarnatiharness** [comment]
trigger codebasehash

**Comment by vivekkarnatiharness** [comment]
trigger codebasehash

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 10 of 32 files at r1, 9 of 34 files at r2, 1 of 3 files at r3, 10 of 18 files at r5, 3 of 14 files at r6, 1 of 1 files at r7, 1 of 2 files at r8, 2 of 7 files at r10, 9 of 13 files at r11, 1 of 1 files at r12, 3 of 4 files at r13, 2 of 2 files at r14, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35669)* status: 1. unresolved discussions. Waiting on team(s): sto.

**Comment by puthrayaharness** [comment]
trigger codebasehash

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r14.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35669)* status: 1. unresolved discussions. Waiting on team(s): sto.

**Pull request merged**